src/main/java/com/minecolonies/coremod/colony/ColonyManager.java

```diff
@@ -483,8 +522,23 @@ public static Colony getColony(@NotNull final World w, @NotNull final BlockPos p
      */
     public static boolean isTooCloseToColony(@NotNull final World w, @NotNull final BlockPos pos)
     {
+        final IChunkmanagerCapability worldCapability = w.getCapability(CHUNK_STORAGE_UPDATE_CAP, null);
+        if (worldCapability == null)
+        {
+            return true;
+        }
         final Chunk centralChunk = w.getChunkFromBlockCoords(pos);
-        return !centralChunk.getCapability(CLOSE_COLONY_CAP, null).getAllCloseColonies().isEmpty();
+        final IColonyTagCapability colonyCap = centralChunk.getCapability(CLOSE_COLONY_CAP, null);
+        if (colonyCap == null)
+        {
+            return true;
+        }
+        final ChunkLoadStorage storage = worldCapability.getChunkStorage(centralChunk.x, centralChunk.z);
```

Comment:
Does this also clear the CLS from memory., to prevent it form being applied again and again.?

Reply:
I added that getChunkStorage does return remove and not get from the map, good catch